### PR TITLE
Update SQLFluff + pre-commit + other dependencies using requirements.in

### PR DESCRIPTION
**Pull request by gwenwindflower**
This is an update to allow the newest SQLFluff to run in pre-commit and for other linting tasks. It also implements a `requirements.in` file to compile our `requirements.txt` for installing dependencies (this is included in this PR because I did this whilst updating SQLFluff, sorry, not very Cool Girl Atomic Commits of me). 

We decided to implement a `requirements.in` because it's built-in to [`uv`](https://github.com/astral-sh/uv) and provides a better development experience.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,16 +17,16 @@ repos:
...
           [
-            "dbt-metricflow[duckdb,snowflake,postgres]~=0.5.0",
+            "dbt-metricflow[snowflake,postgres]~=0.5.0",
             "sqlfluff-templater-dbt~=2.3.5",
           ]
       - id: sqlfluff-fix
         additional_dependencies:
           [
-            "dbt-metricflow[duckdb,snowflake,postgres]~=0.5.0",
+            "dbt-metricflow[snowflake,postgres]~=0.5.0",
```
Comment (dave-connors-3) [inline .pre-commit-config.yaml:26]: that's fun, i didn't know you could use commas here!

File: .sqlfluff [hunk 2]
```diff
@@ -1,18 +1,21 @@
 [sqlfluff]
+templater = jinja
```
Comment (dave-connors-3) [inline .sqlfluff:2]: what's the motivation for this change?

File: packages.yml [hunk 3]
```diff
@@ -1,5 +0,0 @@
-packages:
```
Comment (dave-connors-3) [inline packages.yml:1]: i think we may have a util or two in here, does this need to be deleted?

**Review by dave-connors-3 — COMMENTED** [review]
few comments!

File: packages.yml (hunk 3 above)
Comment (gwenwindflower) [reply packages.yml:1]: oh that's a mistake from splitting them, no this should stay

File: .sqlfluff (hunk 2 above)
Comment (gwenwindflower) [reply .sqlfluff:2]: long story, but i'm going to cut it back to dbt-templater i just wanted to get the PRs reconciled -- doing that now

**Review by dave-connors-3 — APPROVED** [review]

**Review by dave-connors-3 — APPROVED** [review]
🦞

**Review by dave-connors-3 — DISMISSED** [review]
